include/rmm/device_buffer.hpp

```diff
@@ -392,4 +392,29 @@ class device_buffer {
       mr::get_default_resource()};  ///< The memory resource used to
                                     ///< allocate/deallocate device memory
 };
+
+/**--------------------------------------------------------------------------*
+ * @brief Copies rmm::device_buffer to a preallocated host buffer.
+ *
+ * Copies device memory asynchronously on the specified stream
+ *
+ * @throws std::runtime_error if `hb` is `nullptr` or copy fails
+ *
+ * @param db `rmm::device_buffer` to copy to host
+ * @param hb host allocated buffer to copy data to
+ * @param stream CUDA stream on which the device to host copy will be performed
+ *-------------------------------------------------------------------------**/
+void copy_to_host(const device_buffer& db, void* hb, cudaStream_t stream = 0) {
```

Comment:
There were linking errors downstream as this was not declared `inline` and the function source code was not in a `.cpp` file. This has since been fixed with PR ( https://github.com/rapidsai/rmm/pull/230 ).